python/examples/example_errors.ipynb

```diff
@@ -0,0 +1,977 @@
+    "amici_solver = amici_model.getSolver()\n",
+    "amici_solver.setMaxSteps(10 * amici_solver.getMaxSteps())\n",
+    "\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True,\n",
+    "    solver=amici_solver\n",
+    ")\n",
+    "\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])\n",
+    "assert all(rdata.status == amici.AMICI_SUCCESS for rdata in res[RDATAS])\n",
+    "print(\"Simulations finished succesfully.\")\n",
+    "print()\n",
+    "\n",
+    "\n",
+    "# let's relax the relative error tolerance by a factor of 50\n",
+    "print(\"Relaxing relative error tolerance ...\")\n",
+    "amici_solver = amici_model.getSolver()\n",
+    "amici_solver.setRelativeTolerance(50 * amici_solver.getRelativeTolerance())\n",
+    "\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True,\n",
+    "    solver=amici_solver\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])\n",
+    "assert all(rdata.status == amici.AMICI_SUCCESS for rdata in res[RDATAS])\n",
+    "print(\"Simulations finished succesfully.\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "18fd3fa6",
+   "metadata": {},
+   "source": [
+    "## `Internal t = [...] and h = [...] are such that t + h = t on the next step`\n",
+    "\n",
+    "Let's run a simulation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f78179dc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "petab_problem = benchmark_models_petab.get_problem(\"Crauste_CellSystems2017\")\n",
+    "amici_model = import_petab_problem(petab_problem, verbose=False)\n",
+    "\n",
+    "np.random.seed(1)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "53e4b822",
+   "metadata": {},
+   "source": [
+    "**What happened?**\n",
+    "\n",
+    "The forward simulation failed because AMICI the solver exceeded the maximum number of steps. Unlike in the previous case of  `mxstep steps taken before reaching tout` (see above), here we got several additional warnings that the current step size $h$ is numerically zero.\n",
```

Comment:
maybe note that having the current step size $h$ equal to zero is not an issue by itself and that the solver can recover from that. But here the solver does get stuck which results in exceeding the max step number.